crates/shell/src/shells/fish.rs

```diff
@@ -74,7 +133,7 @@ mod tests {
     fn formats_env_var() {
         assert_eq!(
             Fish.format_env_set("PROTO_HOME", "$HOME/.proto"),
-            r#"set -gx PROTO_HOME "$HOME/.proto";"#
+            r#"set -gx PROTO_HOME "\$HOME/.proto";"#
```

Comment:
Does this $ escape break interpolation?

Reply:
No IMO. It is the intended behaviour as I referred in docs. Based on the Fish shell documentation, the `$` character needs to be escaped (`\$`) within double quotes to prevent variable interpolation.